test/cypress/integration/amount/amount.cy.test.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const productStubs = require('../../stubs/products-stubs')
+const serviceStubs = require('../../stubs/service-stubs')
+
+const gatewayAccountId = 666
+const productExternalId = 'a-product-id'
+
+describe('Amount page', () => {
+  describe('when the Payment Link has no price', () => {
+    beforeEach(() => {
+      cy.task('setupStubs', [
+        productStubs.getProductByExternalIdStub({
+          external_id: productExternalId,
+          reference_enabled: true,
+          reference_label: 'invoice number',
+          type: 'ADHOC'
+        }),
+        serviceStubs.getServiceSuccess({
+          gatewayAccountId: gatewayAccountId,
+          serviceName: {
+            en: 'Test service name'
+          }
+        })
+      ])
+
+      Cypress.Cookies.preserveOnce('session')
+    })
+
+    it('should redirect to the `Amount` page', () => {
+      cy.visit('/pay/a-product-id/amount')
+
+      cy.get('[data-cy=back-link]').should('have.attr', 'href', '/pay/a-product-id/reference')
+      cy.get('[data-cy=label]').should('contain', 'Enter amount to pay')
+      cy.get('[data-cy=button]').should('exist')
+    })
+
+    it('when the amount is in the wrong format, should display an error', () => {
+      cy.get('[data-cy=input]').type('invalid amount', { delay: 0 })
+      cy.get('[data-cy=button]').click()
+
+      cy.get('[data-cy=error-summary] a').should('contain', 'Enter an amount in pounds and pence using digits and a decimal point. For example “10.50”')
```

Comment:
```suggestion
      cy.get('[data-cy=error-summary] a')
      .should('contain', 'Enter an amount in pounds and pence using digits and a decimal point. For example “10.50”')
      .should('have.attr', 'href', '#payment-amount') 
```

You could chain these to make this a bit more DRY.

Reply:
That's a good idea - code looks clearer as well.
I will do this.